pkg/operation/botanist/component/etcdcopybackupstask/etcdcopybackupstask.go

```diff
@@ -0,0 +1,207 @@
+		waitTimeout,
+		&druidv1alpha1.EtcdCopyBackupsTask{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      values.Name,
+				Namespace: values.Namespace,
+			},
+		},
+	}
+}
+
+// Deploy creates the EtcdBackupsCopyTask resource. If it already exists, it is deleted and then recreated.
+func (e *etcdCopyBackupsTask) Deploy(ctx context.Context) error {
+	if err := e.Destroy(ctx); err != nil {
+		return err
+	}
+
+	e.task.Spec.MaxBackupAge = e.values.MaxBackupAge
+	e.task.Spec.MaxBackups = e.values.MaxBackups
+	e.task.Spec.SourceStore = e.values.SourceStore
+	e.task.Spec.TargetStore = e.values.TargetStore
+	return e.client.Create(ctx, e.task)
+}
+
+// Wait waits until the EtcdBackupsCopyTask is ready
+func (e *etcdCopyBackupsTask) Wait(ctx context.Context) error {
+	if err := extensions.WaitUntilObjectReadyWithHealthFunction(
+		ctx,
+		e.client,
+		e.logger,
+		WaitForConditions,
+		e.task,
+		"EtcdCopyBackupsTask",
+		e.waitInterval,
+		e.waitTimeout,
+		e.waitTimeout,
+		e.CheckConditions,
+	); err != nil {
+		e.logger.Error(err)
+		return err
+	}
+	return nil
+}
+
+// Destroy deletes the EtcdBackupsCopyTask resource.
+func (e *etcdCopyBackupsTask) Destroy(ctx context.Context) error {
+	return kutil.DeleteObject(
+		ctx,
+		e.client,
+		e.task,
+	)
+}
+
+// WaitCleanup is not implemeted yet.
+func (e *etcdCopyBackupsTask) WaitCleanup(_ context.Context) error { return nil }
+
+// SetSourceStore sets the specifications for the object store provider from which backups will be copied
+func (e *etcdCopyBackupsTask) SetSourceStore(store druidv1alpha1.StoreSpec) {
+	e.values.SourceStore = store
+}
+
+// SetTargetStore sets the specifications for the object store provider to which backups will be copied
+func (e *etcdCopyBackupsTask) SetTargetStore(store druidv1alpha1.StoreSpec) {
+	e.values.TargetStore = store
+}
+
+// WaitForConditions waits until the EtcdCopyBackupsTask conditions have been populated by the etcd druid
+func WaitForConditions(obj client.Object) error {
+	etcdCopyBackupsTask, ok := obj.(*druidv1alpha1.EtcdCopyBackupsTask)
+	if !ok {
+		return fmt.Errorf("expected *duridv1alpha1.EtcdCopyBackupsTask but got %T", obj)
+	}
+	if etcdCopyBackupsTask.DeletionTimestamp != nil {
+		return fmt.Errorf("unexpectedly has a deletion timestamp")
+	}
+
+	generation := etcdCopyBackupsTask.Generation
+	observedGeneration := etcdCopyBackupsTask.Status.ObservedGeneration
+	if observedGeneration == nil {
+		return fmt.Errorf("observed generation not recorded")
+	}
+	if generation != *observedGeneration {
+		return fmt.Errorf("observed generation outdated (%d/%d)", *observedGeneration, generation)
+	}
+
+	if etcdCopyBackupsTask.Status.LastError != nil {
+		return fmt.Errorf("error during reconciliation: %s", *etcdCopyBackupsTask.Status.LastError)
+	}
+
+	for _, condition := range etcdCopyBackupsTask.Status.Conditions {
+		if condition.Type == druidv1alpha1.EtcdCopyBackupsTaskSucceeded ||
+			condition.Type == druidv1alpha1.EtcdCopyBackupsTaskFailed {
```

Comment:
I think it would be best to also check the condition status, although it doesn't matter currently in the future it might happen that it's set to something else than `True`.
```suggestion
		if (condition.Type == druidv1alpha1.EtcdCopyBackupsTaskSucceeded || condition.Type == druidv1alpha1.EtcdCopyBackupsTaskFailed) && condition.Status == druidv1alpha1.ConditionTrue  {
```